### PR TITLE
Fixes #94

### LINKED ISSUE #94: EC256 signature is incompatible with RFC-7518
The library creates ES256 JWS signature as DER encoded ASN1 sequence, but it have to be concatenated R and S ECPoints.

See https://tools.ietf.org/html/rfc7518#section-3.4
The problem code is https://github.com/namshi/jose/blob/master/src/Namshi/JOSE/Signer/OpenSSL/PublicKey.php#L25

The right example is
https://github.com/lcobucci/jwt/blob/master/src/Signer/Ecdsa.php#L82

With best wishes, Roman

### PR DESCRIPTION
**Pull request by tillz**

### CODE REVIEW

File: src/Namshi/JOSE/Signer/OpenSSL/PublicKey.php [hunk 1]
```diff
@@ -23,7 +37,19 @@ public function sign($input, $key, $password = null)
...
+        
+        $asn1Decoder = new ASN1();
 
+        $asn1Decoded = $asn1Decoder->decodeBER($signature);
+        $asn1Decoded = $asn1Decoder->asn1map($asn1Decoded[0], self::$asn1Schema);
+        if( isset($asn1Decoded['r']) && isset($asn1Decoded['s']) &&
+            $asn1Decoded['r'] instanceof BigInteger              && 
+            $asn1Decoded['s'] instanceof BigInteger                 ) {
+
+            $signature = $asn1Decoded['r']->toBytes().$asn1Decoded['s']->toBytes();
```
Comment (Spomky) [inline src/Namshi/JOSE/Signer/OpenSSL/PublicKey.php]: You have to add zero padding for each part.

**Comment by Spomky** [comment]
You should add tests to verify the produced signatures are valid.

See https://tools.ietf.org/html/rfc7520#section-4.3 for some examples.

File: src/Namshi/JOSE/Signer/OpenSSL/PublicKey.php (hunk 1 above)
Comment (tillz) [inline src/Namshi/JOSE/Signer/OpenSSL/PublicKey.php]: Yep, thanks. Are you currently working on a PR from your project? If so, I'd not waste my time fixing this.

Comment (Spomky) [inline src/Namshi/JOSE/Signer/OpenSSL/PublicKey.php]: No I am not

**Comment by tillz** [comment]
see #106

**Pull request closed without merging**